### PR TITLE
hebbot: init at 2.1

### PR DESCRIPTION
**Pull request by a-kenji**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Can you try fixing Darwin by adding the security framework?

**Comment by a-kenji** [comment]
Is this correct? I can't test it sadly. I used the dependencies from `gnutls` as an example.

File: pkgs/servers/matrix-hebbot/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+, pkgs
+, stdenv
+, rustPlatform
+, pkg-config
+, cmake
+, openssl
+, autoconf
+, automake
+, withSecurity ? true
+, Security # darwin Security.framework
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-hebbot/default.nix:12]:
```suggestion
, Security
```

File: pkgs/servers/matrix-hebbot/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    owner = "haecker-felix";
+    repo = "hebbot";
+    rev = "v${version}";
+    sha256 = "sha256-zcsoTWpNonkgJLTC8S9Nubnzdhj5ROL/UGNWUsLxLgs=";
+  };
+
+  cargoSha256 = "sha256-ZNETA2JUZCS8/a2oeF+JCGVKbzeyhp51D0BmBTPToOw=";
+
+  nativeBuildInputs = [ pkg-config cmake ] ++
+    lib.optionals (stdenv.isDarwin && !withSecurity) [ autoconf automake ];
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-hebbot/default.nix:29]:
```suggestion
    lib.optionals stdenv.isDarwin [ autoconf automake ];
```

File: pkgs/servers/matrix-hebbot/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  buildInputs = [ openssl ] ++ lib.optional (stdenv.isDarwin && withSecurity) Security;
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-hebbot/default.nix:31]:
```suggestion
  buildInputs = [ openssl ] ++ lib.optional stdenv.isDarwin Security;
```

File: pkgs/servers/matrix-hebbot/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  meta = with lib; {
+    description = "A Matrix bot which can generate \"This Week in X\" like blog posts ";
+    homepage = "https://github.com/haecker-felix/hebbot";
+    changelog = "https://github.com/haecker-felix/hebbot/releases/tag/v${version}";
+    license = with licenses; [ agpl3 ];
+    maintainers = with maintainers; [ a-kenji ];
+  };
+
+}
+
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-hebbot/default.nix:42]:
```suggestion
  };
}
```
please remove the 2nd final new line

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -3986,6 +3986,10 @@ with pkgs;
 
   hebcal = callPackage ../tools/misc/hebcal {};
 
+  hebbot = callPackage ../servers/matrix-hebbot {
+     inherit (darwin.apple_sdk.frameworks) Security;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:3990]:
```suggestion
   inherit (darwin.apple_sdk.frameworks) Security;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (a-kenji) [reply pkgs/top-level/all-packages.nix:3990]: I think splitting the difference here should be correct, right? One space less?

**Pull request merged**